src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CollectionEditor.cs

```diff
@@ -108,9 +108,16 @@ protected virtual object CreateInstance(Type itemType)
             }
         }
 
-        return itemType.UnderlyingSystemType == typeof(string)
-            ? string.Empty
-            : TypeDescriptor.CreateInstance(host, itemType, argTypes: null, args: null)!;
+        if (itemType.UnderlyingSystemType == typeof(string))
+        {
+            return string.Empty;
+        }
+
+        object? o = TypeDescriptor.CreateInstance(host, itemType, argTypes: null, args: null);
+
+        return o is null
+            ? throw new InvalidOperationException($"Neither {nameof(IDesignerHost)} nor {nameof(TypeDescriptor)} can create an instance of {itemType.FullName}.")
```

Comment:
We'd want to localize this since this would surface to users, could you place this string inside of [SR.resx](https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms.Design/src/Resources/SR.resx)? Afterwards this can be changed to the following
```suggestion
            ? throw new InvalidOperationException(string.Format(SR.CollectionEditorCreateInstanceError, nameof(IDesignerHost), nameof(TypeDescriptor), itemType.FullName))
```